Add entitlement assertions for JIT disabling on macOS

darryl-apple opened the pull request:
#### 9cc524ed511e7ec4127c7dda5cef25610542f8ac
<pre>
Add entitlement assertions for JIT disabling on macOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=300127">https://bugs.webkit.org/show_bug.cgi?id=300127</a>
<a href="https://rdar.apple.com/problem/161912390">rdar://problem/161912390</a>

Reviewed by Mark Lam.

Add release assertions to verify that verified-jit and single-jit
entitlements are present when disabling JIT on macOS. This ensures
processes disabling JIT have the required entitlements for proper
JIT region unmapping behavior.

* Source/JavaScriptCore/jit/ExecutableAllocator.cpp:
(JSC::ExecutableAllocator::disableJIT):

Canonical link: <a href="https://commits.webkit.org/301264@main">https://commits.webkit.org/301264@main</a>
</pre>

MenloDorian reviewed (approved):
r=me but let's just make these RELEASE_ASSERT.  There's no perf implication here, and we want to be able to catch this issue even on release builds.

darryl-apple committed:
Add entitlement assertions for JIT disabling on macOS
https://bugs.webkit.org/show_bug.cgi?id=300127
rdar://problem/161912390

Reviewed by Mark Lam.

Add release assertions to verify that verified-jit and single-jit
entitlements are present when disabling JIT on macOS. This ensures
processes disabling JIT have the required entitlements for proper
JIT region unmapping behavior.

* Source/JavaScriptCore/jit/ExecutableAllocator.cpp:
(JSC::ExecutableAllocator::disableJIT):

Canonical link: https://commits.webkit.org/301264@main
